openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -209,16 +224,59 @@ class AccessLevel:
     NO_ACCESS = None
 
 
-def list_libraries_for_user(user):
+def get_libraries_for_user(user):
     """
-    Lists up to 50 content libraries that the user has permission to view.
-
-    This method makes at least one HTTP call per library so should only be used
-    for development until we have something more efficient.
+    Return content libraries that the user has permission to view.
     """
     qs = ContentLibrary.objects.all()
-    filtered_qs = permissions.perms[permissions.CAN_VIEW_THIS_CONTENT_LIBRARY].filter(user, qs)
-    return [get_library(ref.library_key) for ref in filtered_qs[:50]]
+    return permissions.perms[permissions.CAN_VIEW_THIS_CONTENT_LIBRARY].filter(user, qs)
+
+
+def get_metadata_from_index(queryset):
+    """
+    Take a list of ContentLibrary objects and return metadata stored in
+    ContentLibraryIndex.
+    """
+    metadata = None
+    if ContentLibraryIndexer.indexing_is_enabled():
+        try:
+            library_keys = [lib.library_key for lib in queryset]
+            metadata = ContentLibraryIndexer.get_libraries(library_keys)
+        except (LibraryNotIndexedException, KeyError, ConnectionError) as e:
+            log.exception(e)
+
+    # If ContentLibraryIndex is not available, we query blockstore for a limited set of metadata
+    if metadata is None:
+        uuids = [lib.bundle_uuid for lib in queryset]
+        bundles = get_bundles(uuids)
+        bundle_dict = {
+            bundle.uuid: {
+                'uuid': bundle.uuid,
+                'title': bundle.title,
+                'description': bundle.description,
+                'version': bundle.latest_version,
+            }
+            for bundle in bundles
+        }
+        metadata = [bundle_dict[uuid] for uuid in uuids]
+
+    libraries = [
+        ContentLibraryMetadata(
+            key=lib.library_key,
+            bundle_uuid=metadata[i]['uuid'],
+            title=metadata[i]['title'],
+            description=metadata[i]['description'],
+            version=metadata[i]['version'],
+            allow_public_learning=queryset[i].allow_public_learning,
+            allow_public_read=queryset[i].allow_public_read,
+            num_blocks=metadata[i].get('num_blocks', None),
+            last_published=metadata[i].get('last_published', None),
+            has_unpublished_changes=metadata[i].get('has_unpublished_changes', None),
+            has_unpublished_deletes=metadata[i].get('has_unpublished_deletes', None),
```

Comment:
style nit: you can omit `None` as the second argument, as it's already the default return value of `.get`.